### PR TITLE
[11.0][ADD] stock_picking_whole_scrap: New module for scrap all picking products in one step

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @tecnativa

### CODE REVIEW

**Comment by rafaelbn** [comment]
Hello @sergio-teruel  , do you know why runbot is :red_circle:  it look like a warning

`2018-10-24 10:35:43,357 168 WARNING openerp_test odoo.fields: Field wiz.stock.picking.scrap.line.wizard_id with unknown comodel_name 'stock.picking.scrap'`

**Comment by sergio-teruel** [comment]
@rafaelbn changes done... and fixes

File: stock_picking_scrap_quick/views/stock_picking_views.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+<odoo>
+
+    <record id="view_picking_form" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form"/>
+        <field name="arch" type="xml">
+            <field name="state" position="before">
+                <button name="button_scrap_quick" type="object"
+                        string="Scrap All"
+                        attrs="{'invisible': ['|', '|', '&amp;', ('picking_type_code', '!=', 'incoming'), ('state', 'in', ('draft', 'cancel', 'waiting')), '&amp;', ('picking_type_code', '=', 'incoming'), ('state', '!=', 'done'), ('is_locked', '=', False)]}"/>
```
Comment (rousseldenis) [inline stock_picking_scrap_quick/views/stock_picking_views.xml]: Why just incoming pickings ? Scrap can be done on all pickings (normally with internal locations only)

File: stock_picking_scrap_quick/wizards/stock_picking_scrap_quick.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            raise UserError("You may only return one picking at a time!")
+        res = super().default_get(fields)
+        scrap_lines = []
+        picking = self.env['stock.picking'].browse(
+            self.env.context.get('active_id')
+        )
+        if picking:
+            res.update({'picking_id': picking.id})
+            if picking.state != 'done':
+                raise UserError(_("You may only scarp pickings in done state"))
```
Comment (rousseldenis) [inline stock_picking_scrap_quick/wizards/stock_picking_scrap_quick.py]: scrap

File: stock_picking_scrap_quick/wizards/stock_picking_scrap_quick.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            'scrap_location_id': self.scrap_location_id.id,
+            'scrap_qty': scrap_line.quantity,
+        }
+        return vals
+
+    def create_scrap(self):
+        StockScrap = self.env['stock.scrap']
+        new_scraps = StockScrap.browse()
+        for line in self.line_ids.filtered('quantity'):
+            if line.quantity > line.move_line_id.qty_done:
```
Comment (rousseldenis) [inline stock_picking_scrap_quick/wizards/stock_picking_scrap_quick.py]: Use float_compare

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review. Minor things.

Maybe the name '...quick' does not mean really something functionally. stock_whole_picking_scrap or ... is maybe more relevant

**Comment by pedrobaeza** [comment]
I like the suggestion about the name of @rousseldenis, but the proper name following usual scheme will be `stock_picking_whole_scrap`

File: stock_picking_scrap_quick/views/stock_picking_views.xml (hunk 1 above)
Comment (chienandalu) [reply stock_picking_scrap_quick/views/stock_picking_views.xml]: It seems to be the same domain that the standard scrap button uses, so it should be available in the same cases.

**Review by chienandalu — APPROVED** [review]
Functional test is ok :+1:

**Comment by sergio-teruel** [comment]
@rousseldenis Changes done. About domain, I have kept the same domain as original scrap button.

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1: 

Please @rousseldenis could check that your comments were attended/answered? Thanks!

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@sergio-teruel I will try to pay attention to category entry in manifest (should be 'Warehouse'). It should be coherent in OCA shop interface.

**Comment by pedrobaeza** [comment]
I have changed it by direct commit.

**Pull request merged**